app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdScreenController.kt

```diff
@@ -95,4 +98,30 @@ class ScanSimpleIdScreenController @Inject constructor(
 
     return openPatientSummary.mergeWith(openAddIdToPatientSearchScreen)
   }
+
+  private fun handleShortCodeSearch(events: Observable<UiEvent>): Observable<UiChange> {
+    val hideValidationErrors = events
+        .ofType<ShortCodeChanged>()
+        .map { { ui: Ui -> ui.hideShortCodeValidationError() } }
+
+    val shortCodes = events
+        .ofType<ShortCodeSearched>()
+        .map { it.shortCode }
+        .share()
+
+    val showValidationErrors = shortCodes
+        .filter { !it.isValid() }
+        .map { { ui: Ui -> ui.showShortCodeValidationError() } }
+
+    val openPatientSearchScreenChanges = shortCodes
```

Comment:
Rename this to `openPatientSearchScreen`?

Reply:
Changed on the upcoming PR branch.